python310Packages.yaramod: init at 3.12.2

msm-code opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

YARA is a language used for creating signatures, especially for malware.
Yaramod is a library created by avast, for parsing and working with yara rules (signatures).
The main project page is https://github.com/avast/yaramod

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Commit:
python38Packages.yaramod: init at 3.12.2

Co-authored-by: Robert Schütz <github@dotlambda.de>
Co-authored-by: Fabian Affolter <mail@fabian-affolter.ch>

fabaff commented:
Result of `nixpkgs-review pr 211504` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python311Packages.yaramod</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>python310Packages.yaramod</li>
  </ul>
</details>

fabaff commented:
Depends on #211703

msm-code commented:
Sorry for obvious mistakes - I've fixed the description and changelog locally but apparently didn't push my ammended commit (or forgot `--force`).

Don't know how I missed the build fail for 3.11 - thanks!

dotlambda commented on pkgs/development/python-modules/yaramod/default.nix:
```diff
@@ -0,0 +1,49 @@
+    description = " Parsing of YARA rules into AST and building new rulesets in C++";
+    homepage = "https://github.com/avast/yaramod";
+    changelog = "https://github.com/avast/yaramod/blob/v${version}/CHANGELOG.md";
+    license = with licenses; [ mit ];
```
```suggestion
    license = licenses.mit;
```

dotlambda commented on pkgs/development/python-modules/yaramod/default.nix:
```diff
@@ -0,0 +1,49 @@
+  ];
+
+  meta = with lib; {
+    description = " Parsing of YARA rules into AST and building new rulesets in C++";
```
```suggestion
    description = "Parsing of YARA rules into AST and building new rulesets in C++";
```

dotlambda commented on pkgs/development/python-modules/yaramod/default.nix:
```diff
@@ -0,0 +1,49 @@
+    rev = "refs/tags/v${version}";
+    hash = "sha256-1q+lzNtm8qko9ZAxJjkmPOQjHD5GxB6YyEz0tr+QWGk=";
+  };
+
```
```suggestion

  postPatch = ''
    rm -r deps
  '';

```
We try not to use vendored dependencies.

dotlambda commented on pkgs/development/python-modules/yaramod/default.nix:
```diff
@@ -0,0 +1,49 @@
+  nativeBuildInputs = [
+    cmake
+  ];
+
```
```suggestion

  setupPyBuildFlags = [
    "--with-unit-tests"
  ];

```

dotlambda reviewed (changes requested):
Please squash.

msm-code commented:
Thanks for valuable comments. Squashed and pushed the results.

In the current state of this PR, the build fails. I'm in the process of reworking it to build without the vendored dependencies.

msm-code commented:
I've rewritten the derivation to remove the vendored deps, and re-downloaded them myself.

I've tried to use versions already in nixpkgs, but the code seems to depend on the specific vendored versions (or in one case even patches them). For this reason this modification feels fragile to me (what if the project updates the vendored sources). But if that's the preferred solution, I'm OK with it.

I've tested the code with nixpkgs-review and in a nix-shell. I've also squashed my commit. Asked @dotlambda for re-review.

msm-code commented:
Hi @dotlambda, what do you think about the current version?

I'll add once again that my preference would be to use vendored dependencies (to reduce the maintanance overhead when they are updated in the project - they are pinned to specific version and the project doesn't work with the version we have in nixpkgs).

Nevertheless, I've removed vendored deps as suggested. Let me know if you see anything else that can be improved.

SuperSandro2000 commented on pkgs/development/python-modules/yaramod/default.nix:
```diff
@@ -0,0 +1,89 @@
+    rev = "b09bbf9cea573ee62aab7eccda896e37961d16cd";
+    hash = "sha256-El4WA92t2O/L4wUqH6Xj8w+ANtb6liRwafDhqn8jxjQ=";
+  };
+  pybind11 = fetchFromGitHub {
+    owner = "pybind";
+    repo = "pybind11";
+    rev = "refs/tags/v2.9.2";
+    hash = "sha256-O3bkexUBa+gfiJEM6KSR8y/iVqHqlCFmz/9EghxdIpw=";
+  };
+  json = fetchFromGitHub {
+    owner = "nlohmann";
+    repo = "json";
+    rev = "refs/tags/v3.9.1";
+    hash = "sha256-THordDPdH2qwk6lFTgeFmkl7iDuA/7YH71PTUe6vJCs=";
+  };
```
Why are we not using nixpkgs versions of json and pybind11?

msm-code replied:
Honestly, not sure about pybind11. I tested it earlier and it was broken with nixpkgs vesion of pybind11. Now I've changed it to the nixpkgs version, and it passes all my tests (and the included pytest tests). So probably I was confused/tested a wrong version.

For json I didn't know how to get the "single header include" for it, but found an example in another package from nixpkgs, so I replaced that.

msm-code commented:
Result of `nixpkgs-review pr 211504` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.yaramod</li>
    <li>python311Packages.yaramod</li>
  </ul>
</details>

SuperSandro2000 commented:
@ofborg build python310Packages.yaramod
